src/views/Login.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+    <div class="login-page">
+        <div class="form">
+            <form class="login-form" v-on:submit.prevent>
+                <input type="text" placeholder="username" v-model="username"/>
+                <input type="password" placeholder="password" v-model="password"/>
+                <button @click="doLogin" :disabled=isLoading>login</button>
+            </form>
+            <span v-if="isError" style="color: red">Failed to log in</span>
+        </div>
+    </div>
+</template>
+
+<script>
+export default  {
+    computed: {
+        isLoading() {
+            return this.$store.getters.isLoading;
+        },
+        isError() {
+            return this.$store.getters.isError;
+        },
+    },
+    data() {
+        return {
+            username: '',
+            password: '',
+            error: undefined,
```

Comment:
Currently not using the error for anything. The template uses isError which comes from global state. I could maybe Just dispatch the error to the store.

I feel this model is quite different from react and redux. And i have to get used to it.

Reply:
Could i maybe handle all error cases in the global store? And drop the catch block below?